File: ddd/tests/fixtures/operation-error-set/typescript-class-workspace/billing-domain/src/extra.ts
/** Each operation adds one case that no operation of the mapping names. */

import type { Result } from "./result.ts";

export type IssueInvoiceError = "already-issued" | "empty-lines" | "locked";
export type OpenInvoiceError = "negative-amount" | "missing-customer" | "expired";

export class Invoice {
  static open(): Result<Invoice, OpenInvoiceError> {
    return { ok: true, value: new Invoice() };
  }

  issue(): Result<void, IssueInvoiceError> {
    return { ok: true, value: undefined };
  }
}

File: ddd/tests/fixtures/operation-error-set/rust-workspace/billing-domain/src/extra.rs
//! Each operation adds one case that no operation of the mapping names.

pub enum IssueInvoiceError {
    AlreadyIssued,
    EmptyLines,
    Locked,
}

pub enum OpenInvoiceError {
    NegativeAmount,
    MissingCustomer,
    Expired,
}

pub struct Invoice;

impl Invoice {
    pub fn issue(&mut self) -> Result<(), IssueInvoiceError> {
        Ok(())
    }

    pub fn open() -> Result<Self, OpenInvoiceError> {
        Ok(Self)
    }
}
